docs/content/direct/examples.md

```diff
@@ -60,20 +60,14 @@ manager which connects to the `wds1` front-end and the `imbs1` OCM control plane
    kflex create wds1 -p kubestellar
    ```
 
-1. Run the OCM based transport controller as executable process.  
-**NOTE**: This is work in progress, in the future the controller will be deployed through a Pod and optionally a Helm chart.
-
-   OCM based transport controller is built in a different package, so we need to download the package and run the executable as background process.
+1. Run the OCM based transport controller in a pod.  
+**NOTE**: This is work in progress, in the future the controller will be deployed through a Helm chart.
 
+   Run [transport deployment script](../../../hack/deploy-transport-controller.sh).  
+   This script expects to get three arguments - (1) wds name; (2) imbs name; and (3) transport controller image.  
+   For example, one can deploy transport controller using the following command:
    ```shell
-   OS=$(go env GOOS)
-   ARCH=$(go env GOARCH)
-   OCM_TRANSPORT_PLUGIN_RELEASE="0.1.0-rc2"
-   OCM_TRANSPORT_PLUGIN_TAR=ocm-transport-plugin_${OCM_TRANSPORT_PLUGIN_RELEASE}_${OS}_${ARCH}.tar.gz
-   wget https://github.com/kubestellar/ocm-transport-plugin/releases/download/v${OCM_TRANSPORT_PLUGIN_RELEASE}/${OCM_TRANSPORT_PLUGIN_TAR}
-   tar -xvf ${OCM_TRANSPORT_PLUGIN_TAR} transport-controller
-   rm -fr bin ${OCM_TRANSPORT_PLUGIN_TAR}
-   ./transport-controller --transport-context imbs1 --wds-context wds1 --wds-name wds1 &> transport.log &
+   bash <(curl -s https://raw.githubusercontent.com/kubestellar/kubestellar/main/hack/deploy-transport-controller.sh) wds1 imbs1 ghcr.io/kubestellar/ocm-transport-plugin/transport-controller:0.1.0-rc4
```

Comment:
Instead of `main` this should reference the next release, which will be `0.21.0-rc1`. You are introducing another self-reference here.

Reply:
The main problem in (B) is that curl will fetch only one file, deploy-transport-controller.sh; the user does not have a local copy of the file referenced from deploy-transport-controller.sh.

Reply:
The thing that seems odd to me about requiring a user of a particular release of ks/ks to specify the release of ks/OTP to use is that there is already an answer to "which OCP release?" given if we are following the staggered release approach to coping with the dependency cycle (the synchronized release approach also gives an answer).

Reply:
I was not thinking of _adding_ another self-reference, I was thinking of _moving_ the one new one from examples.md to deploy-transport-controller.sh.

Reply:
> The main problem in (B) is that curl will fetch only one file, deploy-transport-controller.sh; the user does not have a local copy of the file referenced from deploy-transport-controller.sh.

I get your point. that's not so trivial to do the same logic in a single file.
I fixed the other comment but haven't pushed it yet cause I'm thinking about a simple way to answer your comment.

Reply:
I am not sure I understand the difficulty. I would just make deploy-transport-controller.sh do `kubectl apply -f -<<EOF` and line the template.

Reply:
the difficulty is the config map which uses `"$1`, `$2`, etc.. if that's is done in the same file, the bash replaces these placeholders with the first arg of the script (WDS_NAME, IMBS_NAME) and let empty the cpname, etc.. 
that breaks the config map behavior.

in the deployment itself there is no problem.
I'm trying to work around the configmap.

Reply:
try to this locally and see what happens:
```
cat <<EOF
apiVersion: v1
kind: ConfigMap
metadata:
  name: transport-controller-config
data:
  get-kubeconfig.sh: |
    #!/bin/bash
    # this script receives a ControlPlane name and a parameter
    # that determines whether to extract the ControlPlane's in-cluster kubeconfig
    # or the external kubeconfig (if set to "true", the first will be retrieved).
    # The function returns the requested kubeconfig's content in base64.
    # it assumes the kubeconfig context is set to access the hosting cluster.

    cpname="$1"
    get_incluster_key="$2"

    key=""
    if [[ "$get_incluster_key" == "true" ]]; then
      key=$(kubectl get controlplane $cpname -o=jsonpath='{.status.secretRef.inClusterKey}');
    else
      key=$(kubectl get controlplane $cpname -o=jsonpath='{.status.secretRef.key}');
    fi

    # get secret details
    secret_name=$(kubectl get controlplane $cpname -o=jsonpath='{.status.secretRef.name}')
    secret_namespace=$(kubectl get controlplane $cpname -o=jsonpath='{.status.secretRef.namespace}')

    # get the kubeconfig in base64
    kubectl get secret $secret_name -n $secret_namespace -o=jsonpath="{.data.$key}"
EOF
```

Reply:
fixed with escaping.
at first I confused the escaping character `/` with `\` so it didn't work. 
I guess I was too tired and had to sleep in order to get this fixed :).